### ISSUE TITLE
Cross device link error when using LVM partitions

### ISSUE THREAD

**Issue by cgmcintyr** [suzaku/shonenjump#3]
My machine is partitioned using LVM with the following layout:

``` 
 LV     VG     Attr       LSize   Pool Origin Data%  Meta%  Move Log Cpy%Sync Convert
 home   noe-vg -wi-ao---- 150.00g                                                    
 root   noe-vg -wi-ao---- <24.38g                                                    
 swap_1 noe-vg -wi-a----- <15.41g                                                    
 tmp    noe-vg -wi-ao---- 380.00m                                                    
 var    noe-vg -wi-ao----  32.79g               
```

The `os.Rename` function in `entry.go` always returns a "cross device link error" when trying to rename a shonenjump file `/tmp/<file>` to `/home/<user>/.local/share/shonenjump/<file>`.

**Comment by cgmcintyr** [comment]
Here's my proposed fix:
https://github.com/cgmcintyr/shonenjump/commit/40f15514edb4a0251f09a14c8f6102cea2ffcccf

If it looks good, I'll submit a PR :)

**Comment by suzaku** [comment]
@cgmcintyr  

Thanks for reporting.

In your fix, the program fell back to normal copying when `rename` failed.
But `rename` was guaranteed to be atomic why normal copying was not.
I think we can fix the problem by creating the temporary file in the same folder as `path`, thus living in the same file system.

What do you think?

**Comment by cgmcintyr** [comment]
Makes sense - was there any reason for using `/tmp` over creating a temporary file in the same directory as `path`?

**Comment by suzaku** [comment]
@cgmcintyr 

I've fixed this bug in [c8347f2](https://github.com/suzaku/shonenjump/commit/c8347f22d987be8321657e97900a7567d11b67aa).
Thanks again for reporting.
